pkg/traceability/sampling/sampling_test.go

```diff
@@ -192,6 +193,64 @@ func TestShouldSample(t *testing.T) {
 				PerAPI:     true,
 			},
 		},
+		{
+			name: "50% of subscription transactions with PerAPI enabled",
+			apiTransactions: map[string]int{
+				"id1": 50, // expect 50
+				"id2": 50, // expect 50
+				"id3": 50, // expect 50
+				"id4": 50, // expect 50
+			},
+			subIDs: map[string]string{
+				"id1": "sub1",
+				"id2": "sub2",
+				"id3": "sub3",
+				"id4": "sub4",
+			},
+			expectedSampled: 200,
```

Comment:
Why is the expectedSampled at 200 when the percent is 50 and the total requests is 200? I copied this from another test which is set up the same way.

Reply:
4 ids and we are sampling by id. So each of them will be guaranteed 50 of every 100 transactions

Reply:
The way our sampling works, with the counters, the first 50 of each id will be sampled, the next 50 of each id will not be.  So if we then set all ids to send 75 transactions (300 total) we would still expect 200 sampled.